src/main/webapp/app/exercises/text/assess-new/text-assessment-area/text-assessment-area.component.ts

```diff
@@ -27,11 +37,23 @@ export class TextAssessmentAreaComponent implements OnChanges {
     @Input() textBlockRefs: TextBlockRef[];
     @Output() textBlockRefsChange = new EventEmitter<TextBlockRef[]>();
     selectedRef: TextBlockRef | null = null;
+    wordCount = 0;
+    characterCount = 0;
+
+    constructor(private stringCountService: StringCountService) {}
 
     /**
      * Life cycle hook to indicate component change
      */
-    ngOnChanges(): void {
+    ngOnChanges(changes: { [propKey: string]: SimpleChange }): void {
+        for (const propName in changes) {
```

Comment:
Instead of looping over the changes, you can do `if (changes.submission)` and then access the current value via `changes.submission.currentValue`.

Reply:
It could look like this:

```typescript
if (changes.submission) {
    const { text } = changes.submission.currentValue as TextSubmission;
    this.wordCount = this.stringCountService.countWords(text);
    this.characterCount = this.stringCountService.countCharacters(text);
}
```

Reply:
Looks much better. I also always forget about this nice destructuring :) Changed it in a337fc464e6df41d77a4d12374e7066b5c024882